dev/io.openliberty.microprofile.reactive.messaging.internal/src/io/openliberty/microprofile/reactive/messaging/internal/RMAsyncProviderFactoryImpl.java

```diff
@@ -130,6 +130,16 @@ private String getContextServiceName(Map<String, Object> properties) {
         }
     }
 
+    /**
+     * Returns whether a context service is one defined within an application using {@code @ContextServiceDefinition}.
+     *
+     * @param properties the service properties of the {@code WSContextService} service to check
+     * @return {@code true} if the service properties are for an application-defined context service, otherwise {@code false}
+     */
+    private boolean isApplicationDefinedContextService(Map<String, Object> properties) {
+        return properties.containsKey("application") && properties.containsKey("module");
```

Comment:
Identify application-defined context services in java:global, java:app, and java:comp as well as those in java:module
```suggestion
        return !"file".equals(properties.get("config.source"));
```

Reply:
Thanks, I'll squash this in